go/rtl/auth.go

```diff
@@ -123,16 +128,24 @@ func (s *AuthSession) Do(result interface{}, method, ver, path string, reqPars m
 		}
 	}
 
-	// create request context with timeout
-	var timeoutInSeconds int32 = 120 //seconds
+	parent := context.Background()
+	if s.Config.Context != nil {
+		parent = s.Config.Context
+	}
+	if options != nil && options.Context != nil {
+		parent = options.Context
+	}
+
+	// create request context with timeout from options or else config or else 120 seconds
+	var timeoutInSeconds int32 = 120
 	if s.Config.Timeout != 0 {
 		timeoutInSeconds = s.Config.Timeout
 	}
 	if options != nil && options.Timeout != 0 {
 		timeoutInSeconds = options.Timeout
 	}
 
-	ctx, cncl := context.WithTimeout(context.Background(), time.Second*time.Duration(timeoutInSeconds))
+	ctx, cncl := context.WithTimeout(parent, time.Second*time.Duration(timeoutInSeconds))
```

Comment:
I think an easier to understand behavior would be:
If you use config context or options context, the timeout defined in options or config is ignored. 

Trying to document/understad the how timeout works with context/config/options is confusing and harder to test.

Reply:
I disagree - see my previous comment. I actually think this is _easier_ to document and the tests are already in place.